packages/manager/apps/dedicated/client/app/components/ip/agoraOrder/ip-ip-agoraOrder.controller.js

```diff
@@ -271,27 +271,29 @@ export default class AgoraIpOrderCtrl {
         );
 
         return ipCountryAvailablePromise
-          .then((countries) => {
-            if (countries && countries.length > 0) {
-              const ipOffersByRegion = AgoraIpOrderCtrl.getRegionsOffers(
-                countries,
-              );
-              this.failoverIpOffers = this.getOfferDetails(
-                failoverIpOfferDetails,
-                ipOffersByRegion,
-                countries,
-              );
-              this.blockIpOffers = this.getOfferDetails(
-                blockIpOfferDetails,
-                ipOffersByRegion,
-                countries,
-              ).sort((a, b) => a.price.value - b.price.value);
-            } else {
-              this.ipOffers = AgoraIpOrderCtrl.filterOfferDetailsFromServiceName(
-                ipOfferDetails,
+          .then((data) => {
+            let countries = data;
+            if (data.length === 0) {
+              const REGION = AgoraIpOrderCtrl.getRegionFromServiceName(
                 this.model.selectedService.serviceName,
               );
+              countries = IP_LOCATION_GROUPS.find((group) =>
+                group.labels.includes(REGION),
+              )?.countries;
```

Comment:
**mandatory**:
```suggestion
              ).countries;
```
There is a `countries` key in each item of the `IP_LOCATION_GROUPS` array

Reply:
I let the safe traversal operator just in case if the `find` does not match any matching entry.